woocommerce/woocommerce-gateway-stripe
# PI: Do not update an intent's customer after intent creation.

Fixes #1068 .

#### Changes proposed in this Pull Request:

- Reuse the customer, already associated with a PaymentIntent

During the creation of PaymentIntents, they get associated with a Stripe customer, if one is available. Since `4.3.0` this is always the case, because we are generating Stripe customers for every transaction.

If there is a validation error during the confirmation of a payment intent, the current code base creates a new customer (when using guest checkout) and tries to update the payment intent, however the intent is already in a state, which does not allow that update.

**This PR** introduces the `update_customer` method of `WC_Stripe_User`, and the necessary logic to call it.

#### Testing instructions

Test without this branch first:

1. Checkout as guest, using the zip and address line 1 validation error card number: `4000000000000010`, which correctly displays the zip code invalid error
2. Retry payment using a valid card number (i.e. `4242424242424242`)
3. Error shows: "Sorry, we are unable to process your payment at this time. Please retry later."

After switching to the branch, try the same steps. There should be no errors.

-------------------
